test/src/main/java/org/apache/accumulo/test/fate/accumulo/FateMutatorImplIT.java

```diff
@@ -0,0 +1,154 @@
+
+  @AfterAll
+  public static void tearDown() {
+    SharedMiniClusterBase.stopMiniCluster();
+  }
+
+  @Override
+  protected Duration defaultTimeout() {
+    return Duration.ofMinutes(5);
+  }
+
+  @Test
+  public void putRepo() throws Exception {
+    final String table = getUniqueNames(1)[0];
+    try (AccumuloClient client = Accumulo.newClient().from(getClientProps()).build()) {
+      client.tableOperations().create(table, ntc);
+
+      ClientContext context = (ClientContext) client;
+
+      final long tid = RANDOM.get().nextLong() & 0x7fffffffffffffffL;
+
+      // add some repos in order
+      FateMutatorImpl<FateIT.TestEnv> fateMutator = new FateMutatorImpl<>(context, table, tid);
+      fateMutator.putRepo(100, new FateIT.TestRepo("test")).mutate();
+      FateMutatorImpl<FateIT.TestEnv> fateMutator1 = new FateMutatorImpl<>(context, table, tid);
+      fateMutator1.putRepo(99, new FateIT.TestRepo("test")).mutate();
+      FateMutatorImpl<FateIT.TestEnv> fateMutator2 = new FateMutatorImpl<>(context, table, tid);
+      fateMutator2.putRepo(98, new FateIT.TestRepo("test")).mutate();
+
+      // make sure we cant add a repo that has already been added
+      FateMutatorImpl<FateIT.TestEnv> fateMutator3 = new FateMutatorImpl<>(context, table, tid);
+      assertThrows(IllegalStateException.class,
+          () -> fateMutator3.putRepo(98, new FateIT.TestRepo("test")).mutate(),
+          "Repo in position 98 already exists. Expected to not be able to add it again.");
+      FateMutatorImpl<FateIT.TestEnv> fateMutator4 = new FateMutatorImpl<>(context, table, tid);
+      assertThrows(IllegalStateException.class,
+          () -> fateMutator4.putRepo(99, new FateIT.TestRepo("test")).mutate(),
+          "Repo in position 99 already exists. Expected to not be able to add it again.");
+    }
+  }
+
+  @Test
+  public void requireStatus() throws Exception {
+    final String table = getUniqueNames(1)[0];
+    try (AccumuloClient client = Accumulo.newClient().from(getClientProps()).build()) {
+      client.tableOperations().create(table, ntc);
+
+      ClientContext context = (ClientContext) client;
+
+      final long tid = RANDOM.get().nextLong() & 0x7fffffffffffffffL;
+
+      // use require status passing all statuses. without the status column present this should fail
+      FateMutatorImpl<FateIT.TestEnv> fateMutator = new FateMutatorImpl<>(context, table, tid);
+      assertThrows(IllegalStateException.class,
+          () -> fateMutator.requireStatus(ReadOnlyFateStore.TStatus.values())
+              .putStatus(ReadOnlyFateStore.TStatus.NEW).mutate());
+
+      logAllEntriesInTable(table, client); // this prints nothing on the table as we expect
+
+      // use require status without passing any statuses to require that the status column is absent
+      FateMutatorImpl<FateIT.TestEnv> fateMutator0 = new FateMutatorImpl<>(context, table, tid);
+      fateMutator0.requireStatus().putStatus(ReadOnlyFateStore.TStatus.NEW).mutate();
+
```

Comment:
would be good to read the status after each change or attempted change to ensure its as expected

Reply:
49f6f69